src/main/kotlin/com/autonomousapps/tasks/BuildHealthTask.kt

```diff
@@ -0,0 +1,35 @@
+package com.autonomousapps.tasks
+
+import com.autonomousapps.TASK_GROUP_DEP
+import com.autonomousapps.advice.BuildHealth
+import com.autonomousapps.internal.utils.fromJsonList
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+
+abstract class BuildHealthTask : DefaultTask() {
+
+  init {
+    group = TASK_GROUP_DEP
+    description = "Generates holistic advice for whole project, and can fail the build if desired"
```

Comment:
I would suggest to re-phrase so the description is end-user facing rather than internal to plugin.

For instance: `Generates report on the dependencies and plugins (api vs implementation, unsed dependencies, ...) for all projects`